providers/cscglobal/api.go

```diff
@@ -0,0 +1,180 @@
+package cscglobal
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sort"
+)
+
+const apiBase = "https://apis.cscglobal.com/dbs/api/v2"
+
+// Api layer for CSC Global
+
+type api struct {
+	key          string
+	token        string
+	notifyEmails []string
+}
+
+type requestParams map[string]string
+
+type errorResponse struct {
+	Code        string `json:"code"`
+	Description string `json:"description"`
+	Value       string `json:"value,omitempty"`
+}
+
+type nsModRequest struct {
+	Domain        string   `json:"qualifiedDomainName"`
+	NameServers   []string `json:"nameServers"`
+	DNSType       string   `json:"dnsType,omitempty"`
+	Notifications struct {
+		Enabled bool     `json:"enabled,omitempty"`
+		Emails  []string `json:"additionalNotificationEmails,omitempty"`
+	} `json:"notifications"`
+	ShowPrice    bool     `json:"showPrice,omitempty"`
+	CustomFields []string `json:"customFields,omitempty"`
+}
+
+type nsModRequestResult struct {
+	Result struct {
+		Domain string `json:"qualifiedDomainName"`
+		Status struct {
+			Code                  string `json:"code"`
+			Message               string `json:"message"`
+			AdditionalInformation string `json:"additionalInformation"`
+			UUID                  string `json:"uuid"`
+		} `json:"status"`
+	} `json:"result"`
+}
+
+type domainRecord struct {
+	Nameserver []string `json:"nameservers"`
+}
+
+func (c *api) getNameservers(domain string) ([]string, error) {
+	var bodyString, err = c.get("/domains/" + domain)
+	if err != nil {
+		return []string{}, err
+	}
+
+	var dr domainRecord
+	json.Unmarshal(bodyString, &dr)
+	ns := []string{}
+	for _, nameserver := range dr.Nameserver {
+		ns = append(ns, nameserver)
+	}
+	sort.Strings(ns)
+	return ns, nil
+}
+
+func (c *api) updateNameservers(ns []string, domain string) error {
+	req := nsModRequest{}
+	req.Domain = domain
+	req.NameServers = ns
+	req.DNSType = "OTHER_DNS"
+	req.ShowPrice = false
+	if c.notifyEmails != nil {
+		req.Notifications.Enabled = true
+		req.Notifications.Emails = c.notifyEmails
+	} else {
+		req.Notifications.Enabled = false
+	}
+	req.CustomFields = []string{}
+
+	requestBody, err := json.MarshalIndent(req, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	bodyString, err := c.put("/domains/nsmodification", requestBody)
+	if err != nil {
+		return fmt.Errorf("CSC Global: Error update NS : %s", err)
```

Comment:
change `%s` to `%w` 

When using fmt.Errorf() if the variable is an `error`, use `%w` so that the system 'wraps' the error.